sources/HUBViewController.m

```diff
@@ -1407,31 +1407,25 @@ - (void)scrollToRootBodyComponentAtIndex:(NSUInteger)componentIndex
 {
     NSParameterAssert(componentIndex <= (NSUInteger)[self.collectionView numberOfItemsInSection:0]);
 
-    NSIndexPath * const rootIndexPath = [NSIndexPath indexPathForItem:(NSInteger)componentIndex inSection:0];
     CGPoint const contentOffset = [self.scrollHandler contentOffsetForDisplayingComponentAtIndex:componentIndex
                                                                                   scrollPosition:scrollPosition
                                                                                     contentInset:self.collectionView.contentInset
                                                                                      contentSize:self.collectionView.contentSize
                                                                                   viewController:self];
-
-    void (^completionWrapper)() = ^{
-        completion();
-    };
     
-    // If the component is already visible, the completion handler can be called instantly.
-    if ([self.collectionView.indexPathsForVisibleItems containsObject:rootIndexPath]) {
-        [self setContentOffset:contentOffset animated:animated];
-        completionWrapper();
+    // If the target offset is the same, the completion handler can be called instantly.
+    if (CGPointEqualToPoint(contentOffset, self.collectionView.contentOffset)) {
+        completion();
     // If the scrolling is animated, the animation has to end before the new component can be retrieved.
     } else if (animated) {
-        self.pendingScrollAnimationCallback = completionWrapper;
-        [self setContentOffset:contentOffset animated:animated];
+        self.pendingScrollAnimationCallback = completion;
+        [self.collectionView setContentOffset:contentOffset animated:animated];
```

Comment:
Josu is right. It might break this bug fix: https://github.com/spotify/HubFramework/pull/178
Please use `[self setContentOffset:contentOffset animated:animated];` instead.